frontend/src/components/DownloadAnalysisLink/BuildAnalysis.ts

```diff
@@ -0,0 +1,110 @@
+import { kebabCase } from 'lodash'
+import { flattenDeep } from 'lodash'
+import { CONFIG } from '../../config/config'
+import { CENSUS_MAPPING } from '../../constants/census'
+import { Adequacies, AdequacyMode, Dataset, Method, RepresentativePoint } from '../../constants/datatypes'
+import { getLegend } from '../../utils/adequacy'
+import { averageMeasure, maxMeasure, minMeasure } from '../../utils/analytics'
+import { generateCSV } from '../../utils/csv'
+import { adequaciesFromServiceArea, representativePointsFromServiceAreas, summaryStatisticsByServiceArea } from '../../utils/data'
+import { parseSerializedServiceArea } from '../../utils/formatters'
+import { precisionRound } from '../../utils/numbers'
+import { snakeCase } from '../../utils/string'
+
+const staticCsvRootUrl: string = CONFIG.staticAssets.rootUrl
+const staticCsvPath: string = CONFIG.staticAssets.csv.path
+
+const defaultPrecision: number = 2 // Default decimal precision for CSV.
+
+/**
+ * Build an analysis CSV for download from the census data.
+ */
+export function buildCsvFromData(method: Method, serviceAreas: string[], adequacies: Adequacies, representativePoints: RepresentativePoint[], noAlert?: boolean) {
+  let censusCategories = Object.keys(CENSUS_MAPPING).sort()
+  let headers = [
+    'county',
+    'total_' + formatLegend(getLegend(method, AdequacyMode.ADEQUATE_0)),
+    'total_' + formatLegend(getLegend(method, AdequacyMode.ADEQUATE_1)),
+    'total_' + formatLegend(getLegend(method, AdequacyMode.ADEQUATE_2)),
+    'total_' + formatLegend(getLegend(method, AdequacyMode.INADEQUATE)),
+    'min_' + method,
+    'avg_' + method,
+    'max_' + method
+  ]
+
+  headers.push.apply(headers, getHeadersForCensusCategories(method, censusCategories))
+
+  if (!noAlert && serviceAreas.length > 100) {
+    alert('There are many service areas in this state! Preparing the file may take a moment...')
```

Comment:
this was updated in my PR to add a confirm step, looks like it didn't get incorporated?